### ISSUE TITLE
VB: Async method returning tuple with names crashes compiler when producing PDBs

### ISSUE THREAD

**Issue by ayauka** [dotnet/roslyn#18762]
**Version Used**: 
Visual studio 2017 (compiler vbc 2.0.0.61501)

**Steps to Reproduce**:

1. Open attached project: [ValueTuple_Test.zip](https://github.com/dotnet/roslyn/files/930276/ValueTuple_Test.zip)
2. Compile it
3. Checkout error

**Expected Behavior**:
Compile without errors
**Actual Behavior**:
Error on compile

**Comment by jcouv** [comment]
@ayauka I'm trying to repro the issue. Thanks for the attached repro project.
Somehow, I think I'm hitting a different problem than you.

I'm getting error: `1>D:\issues\ValueTuple_Test\ValueTuple_Test\ValueTuple_Test.vbproj(197,5): error : This project references NuGet package(s) that are missing on this computer. Use NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is ..\..\..\..\..\..\packages\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.1.0.3\build\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.props.`

Could you share more details about the error you are seeing? Maybe include a snippet/screenshot? (You can paste it into github)

**Comment by ayauka** [comment]
I will reconfigure the project to work on machines that install packages on one folder before project files. Sorry for that sometimes I forget that not everyone uses sames directories and the packages not everytime considers that some users changes defaults.

Here is updated project: 
[ValueTuple_Test.zip](https://github.com/dotnet/roslyn/files/930276/ValueTuple_Test.zip)

**Comment by jcouv** [comment]
Thanks @ayauka. The new project file repro'ed.

The detailed build log gives some more hints. I'll investigate.
```
1>  Unhandled Exception: System.ArgumentNullException: Value cannot be null. (TaskId:24)
1>  Parameter name: value (TaskId:24)
1>     at System.Reflection.Throw.ArgumentNull(String parameterName) (TaskId:24)
1>     at System.Reflection.Metadata.BlobBuilder.WriteUTF8(String value, Boolean allowUnpairedSurrogates) (TaskId:24)
1>     at Microsoft.Cci.CustomDebugInfoWriter.SerializeTupleElementNames(ArrayBuilder`1 locals, BlobBuilder cmw) (TaskId:24)
1>     at Microsoft.Cci.CustomDebugInfoWriter.SerializeRecord[T](CustomDebugInfoKind kind, T debugInfo, Action`2 recordSerializer) (TaskId:24)
1>     at Microsoft.Cci.CustomDebugInfoWriter.SerializeTupleElementNames(IMethodBody methodBody, ArrayBuilder`1 customDebugInfo) (TaskId:24)
1>     at Microsoft.Cci.CustomDebugInfoWriter.SerializeMethodDebugInfo(EmitContext context, IMethodBody methodBody, Int32 methodToken, Boolean emitEncInfo, Boolean suppressNewCustomDebugInfo, Boolean& emitExternNamespaces) (TaskId:24)
1>     at Microsoft.Cci.PdbWriter.SerializeDebugInfo(IMethodBody methodBody, StandaloneSignatureHandle localSignatureHandleOpt, CustomDebugInfoWriter customDebugInfoWriter) (TaskId:24)
1>     at Microsoft.Cci.MetadataWriter.SerializeMethodBodies(BlobBuilder ilBuilder, PdbWriter nativePdbWriterOpt, Blob& mvidStringFixup) (TaskId:24)
1>     at Microsoft.Cci.MetadataWriter.BuildMetadataAndIL(PdbWriter nativePdbWriterOpt, BlobBuilder ilBuilder, BlobBuilder mappedFieldDataBuilder, BlobBuilder managedResourceDataBuilder, Blob& mvidFixup, Blob& mvidStringFixup) (TaskId:24)
1>     at Microsoft.Cci.PeWriter.WritePeToStream(EmitContext context, CommonMessageProvider messageProvider, Func`1 getPeStream, Func`1 getPortablePdbStreamOpt, PdbWriter nativePdbWriterOpt, String pdbPathOpt, Boolean allowMissingMethodBodies, Boolean isDeterministic, CancellationToken cancellationToken) (TaskId:24)
1>     at Microsoft.CodeAnalysis.Compilation.SerializeToPeStream(CommonPEModuleBuilder moduleBeingBuilt, EmitStreamProvider peStreamProvider, EmitStreamProvider pdbStreamProvider, Func`1 testSymWriterFactory, DiagnosticBag diagnostics, Boolean metadataOnly, CancellationToken cancellationToken) (TaskId:24)
1>     at Microsoft.CodeAnalysis.CommonCompiler.RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, CancellationToken cancellationToken) (TaskId:24)
1>     at Microsoft.CodeAnalysis.CommonCompiler.Run(TextWriter consoleOutput, CancellationToken cancellationToken) (TaskId:24)
1>     at Microsoft.CodeAnalysis.VisualBasic.CommandLine.Vbc.<>c__DisplayClass1_0.<Run>b__0(TextWriter tw) (TaskId:24)
1>     at Microsoft.CodeAnalysis.CommandLine.ConsoleUtil.RunWithUtf8Output[T](Func`2 func) (TaskId:24)
1>     at Microsoft.CodeAnalysis.CommandLine.ConsoleUtil.RunWithUtf8Output[T](Boolean utf8Output, TextWriter textWriter, Func`2 func) (TaskId:24)
1>     at Microsoft.CodeAnalysis.VisualBasic.CommandLine.Vbc.Run(String[] args, BuildPaths buildPaths, TextWriter textWriter, IAnalyzerAssemblyLoader analyzerLoader) (TaskId:24)
1>     at Microsoft.CodeAnalysis.CommandLine.DesktopBuildClient.RunLocalCompilation(String[] arguments, BuildPaths buildPaths, TextWriter textWriter) (TaskId:24)
1>     at Microsoft.CodeAnalysis.CommandLine.BuildClient.RunCompilation(IEnumerable`1 originalArguments, BuildPaths buildPaths, TextWriter textWriter) (TaskId:24)
1>     at Microsoft.CodeAnalysis.CommandLine.DesktopBuildClient.Run(IEnumerable`1 arguments, IEnumerable`1 extraArguments, RequestLanguage language, CompileFunc compileFunc, IAnalyzerAssemblyLoader analyzerAssemblyLoader) (TaskId:24)
1>     at Microsoft.CodeAnalysis.VisualBasic.CommandLine.Program.Main(String[] args, String[] extraArgs) (TaskId:24)
1>     at Microsoft.CodeAnalysis.VisualBasic.CommandLine.Program.Main(String[] args) (TaskId:24)
1>  Microsoft (R) Visual Basic Compiler version 2.0.0.61501 (TaskId:24)
```

**Comment by ayauka** [comment]
That's the problem exactly I have, when I only use literal tuples without names, no problem, but when names used it fails, thanks for attention.

**Comment by jcouv** [comment]
Note to self (since I'll have to table this for a couple of days): https://github.com/jcouv/roslyn/tree/tuple-pdbcrash has repro and fix-in-progress.

**Comment by jcouv** [comment]
Fixed by #18850.
The fix will ship in Visual Studio 2017 version 15.3.
Thanks @ayauka for reporting this issue.
